rfcs/0000-observability-api.md

```diff
@@ -0,0 +1,234 @@
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+In terms of how the information supplied in the Observability CR will get passed to the other Kuadrant components, multiple options have been brought up:
+
+directly - like setting flags or configuration directly on the deployments of the different Kuadrant components.
+indirectly -  Passing the information to Authorino & limitador via the Authorino & Limitador CRs
+indirectly -  Passing the information to Authorino & limitador in the form of there own Observability CRs
+
+The best approach would be the indirect approach, meaning once the Observability CR is updated the information is passed to the relevant component CR. For example the tracing section in the Authorino CR spec would be updated with the required endpoints and other configuration in the Observability CR, this would then be updated in the Authorino CR. This does mean that the spec exposed is whats in the Kuadrant component CR's but new changes can be requested and implemented in said components.
+
+#### Adding, modifying and deleting values or no values
+
+If a values is being configure either being added, modified or removed all changes will have to be made in the Observability CR. If the component level operators are updated to change these values without the Observability CR being changed they will be overridden back to the source of truth the Observability CR provides. 
+
+If the value is removed from the Observability CR it will also be removed from the relevant component CRs. If the value is removed from the component CR but not the Observability CR it will be overridden and added back. 
+
+
+# Unresolved questions
+
+1. Do we want a value where a person just wants everything "on", they want all the observing. Where a value a the root of the observability section can be set to say give me all you got?
+2. Do we go with a observability policy instead of the proposed configuration style CR like whats being suggested? TLDR of what that would look like is quite similar to the observability CR in the fact that the observability piece is pulled out but instead of there just being one source of truth with the current method, and having the fine grained capabilities be in the one location. Instead you would have for most use cases a single observability policy that would be attached at the Kuadrant component level, in there you would have a Kuadrant level view i.e if i enable tracing here its enabled for all the components that can handle it. From there if a component needs some special configuration a separate policy can be created and attached to that specific component instead. 
```

Comment:
@Boomatang Does that cover for the most part what we were discussing this morning?

Reply:
Im now leaning to this solution being the one to go with but want to get other peoples thoughts before i write a full rfc on it

Reply:
I think that covers what we talked about.